sdk/clientcore/core/src/main/java/io/clientcore/core/http/pipeline/HttpInstrumentationPolicy.java

```diff
@@ -203,37 +213,40 @@ public HttpInstrumentationPolicy(HttpInstrumentationOptions instrumentationOptio
     @SuppressWarnings("try")
     @Override
     public Response<?> process(HttpRequest request, HttpPipelineNextPolicy next) {
-        boolean isTracingEnabled = tracer.isEnabled();
-        if (!isTracingEnabled && !isLoggingEnabled) {
+        final boolean isTracingEnabled = tracer.isEnabled();
+        final boolean isMetricsEnabled = meter.isEnabled();
+        if (!isTracingEnabled && !isLoggingEnabled && !isMetricsEnabled) {
             return next.process();
         }
 
         ClientLogger logger = getLogger(request);
         final long startNs = System.nanoTime();
-        String redactedUrl = getRedactedUri(request.getUri(), allowedQueryParameterNames);
-        int tryCount = HttpRequestAccessHelper.getTryCount(request);
+        final String redactedUrl = getRedactedUri(request.getUri(), allowedQueryParameterNames);
+        final int tryCount = HttpRequestAccessHelper.getTryCount(request);
         final long requestContentLength = getContentLength(logger, request.getBody(), request.getHeaders(), true);
 
-        InstrumentationContext instrumentationContext
-            = request.getRequestOptions() == null ? null : request.getRequestOptions().getInstrumentationContext();
-        Span span = Span.noop();
-        if (isTracingEnabled) {
-            if (request.getRequestOptions() == null || request.getRequestOptions() == RequestOptions.none()) {
-                request.setRequestOptions(new RequestOptions());
-            }
-
-            span = startHttpSpan(request, redactedUrl, instrumentationContext);
-            instrumentationContext = span.getInstrumentationContext();
-            request.getRequestOptions().setInstrumentationContext(instrumentationContext);
+        Map<String, Object> metricAttributes = isMetricsEnabled ? new HashMap<>(8) : null;
```

Comment:
Am I reading this right, but could this check lead to a `NullPointerException`?

- Creation of `metricAttributes` is based on `isMetricsEnabled` (`meter.isEnabled()`)
- This is used in `setAttributes` which checks where attributes need to be added with `httpRequestDuration.isEnabled()`

Is it possible that `meter.isEnabled()` and `httpRequestDuration.isEnabled()` are different?

Reply:
simplified it - now relying on isMetricsEnabled instance var everywhere